contributors/design-proposals/vertical-pod-autoscaler.md

```diff
@@ -0,0 +1,486 @@
+   change them during lifetime of the Pod.
+2. `"auto"` (default): VPA assigns resources on Pod creation and
+   additionally can update them during lifetime of the Pod, including evicting /
+   rescheduling the Pod.
+3. `"off"`: VPA never changes Pod resources. The recommender still sets the
+   recommended resources in the VPA object. This can be used for a “dry run”.
+
+To disable VPA updates the user can do any of the following: (1) change the
+updatePolicy to `"off"` or (2) delete the VPA (and remove VPA Initializers,
+in case they were used) or (3) change the Pod labels to no longer match the VPA
+selector.
+
+Note: disabling VPA prevents it from doing further changes, but does not revert
+resources of the running Pods, until they are updated.
+For example, when running a Deployment, the user would need to perform an update
+to revert Pod to originally specified resources.
+
+#### Resource Policy ####
+The resources policy controls how VPA computes the recommended resources.
+In MVP it consists of (optional) lower and upper bound on the request of each
+container.
+The resources policy could later be extended with additional knobs to let the
+user tune the recommendation algorithm to their specific use-case.
+
+#### Recommendation ####
+The VPA resource has an output-only field keeping a recent recommendation,
+filled by the Recommender. Its primary function is to act as a recommendation
+cache, to limit the impact of temporary unavailability of the Recommender.
+
+#### Initializers ####
+[Initializers](https://github.com/kubernetes/kubernetes/issues/3585)
+are a planned feature which will allow different agents to asynchronously
+initialize parts of the resource at creation time.
+Users of VPA will be able (and encouraged) to use Initializers to ensure that
+Pods are not scheduled before a matching VPA is present.
+This allows to avoid race conditions when Pods are submitted before or shortly
+after the VPA object was added, in which case it’s not possible to predict
+whether the Pods will be scheduled using original resources or recommended
+resources.
+Both VPA Admission Controller and VPA Controller will remove the VPA initializer
+after setting Pod resources. Details of this mechanism will be designed when
+Initializers are available.
+
+### Admission Controller ###
+
+VPA Admission Controller intercepts Pod creation requests. If the Pod is matched
+by a VPA config with mode not set to “off”, the controller rewrites the request
+by applying recommended resources to the Pod spec. Otherwise it leaves the Pod
+spec unchanged.
+
+The controller gets the recommended resources by fetching
+/recommendedPodResources from the Recommender. If the call times out or fails,
+the controller falls back to the recommendation cached in the VPA object.
+If this is also not available and the Pod doesn’t have a VPA Initializer, it is
+rejected.
+
+### Life of the Pod ###
+There are two paths for a Pod using VPA to get scheduled.
```

Comment:
This seems unnecessarily complex for the initial implementation. I'd just use one path.